noxfile.py

```diff
@@ -105,3 +106,32 @@ def docs(session: nox.Session) -> None:
         sphinx_args.insert(0, "--open-browser")
 
     session.run(sphinx_cmd, *sphinx_args)
+
+
+# The following sessions are only to be run in CI to check the nox GHA action
+if ON_GITHUB_ACTIONS:
+
+    def _check_python_version(session: nox.Session) -> None:
+        if session.python.startswith("pypy"):
+            python_version = session.python[4:]
+            implementation = "pypy"
+        else:
+            python_version = session.python
+            implementation = "cpython"
+        session.run(
+            "python",
+            "-c",
+            "import sys; assert '.'.join(str(v) for v in sys.version_info[:2]) =="
+            f" '{python_version}'",
+        )
+        if python_version[:2] != "2.":
+            session.run(
+                "python",
+                "-c",
+                f"import sys; assert sys.implementation.name == '{implementation}'",
+            )
+
+    @nox.session(python=["3.7", "3.8", "3.9", "3.10", "pypy3.7", "pypy3.8", "pypy3.9"])
+    def github_actions_default_tests(session: nox.Session) -> None:
+        """Check default versions installed by the nox GHA Action"""
+        _check_python_version(session)
```

Comment:
It's okay to define these in the normal scope, just have a check at the top level where we set `nox.options.sessions` to extend the list to include these when we're on GHA. Similar to how we decide whether to run the conda tests

Reply:
I think there was some misunderstanding here. Indeed the intended original behavior was the one from your edit.
I wrongly assumed you didn't want those sessions to even show up with `nox -l` when run locally.
Will revert this change.

Reply:
Yep that's my bad sorry! I forgot we set the default sessions anyway with `nox.options.sessions` just define these the normal way and then we can call them directly with the `--session` flag in the GHA jobs to test the action 👍🏻